src/LoopInterface.php

```diff
@@ -240,6 +240,34 @@ public function isTimerActive(TimerInterface $timer);
     public function futureTick(callable $listener);
 
     /**
+     * Registers a signal listener with the loop, which
+     * on it's turn registers it with a signal handler
+     * suitable for the loop implementation.
+     *
+     * See also [example #4](examples).
+     *
+     * @param int $signal
+     * @param callable $listener
+     *
+     * @return void
+     */
+    public function onSignal($signal, callable $listener);
```

Comment:
Should probably be named `addSignal()` for consistency with other methods?

Also, the docs should probably include whether multiple listeners can be attached to the same signal?